src/platform/FailSafeContext.cpp

```diff
@@ -45,6 +54,7 @@ void FailSafeContext::FailSafeTimerExpired()
     mFailSafeArmed                  = false;
     mAddNocCommandHasBeenInvoked    = false;
     mUpdateNocCommandHasBeenInvoked = false;
+    ConfigurationMgr().SetFailSafeArmed(false);
```

Comment:
But what happens now if we crash or shut down before the event we just posted is processed?  That will skip the cleanup, no?

Reply:
How? I think we always first get to here before the event is processed asynchronously, right?

Reply:
Yes, the kFailSafeTimerExpired is processed by all  subscribed components asynchronously, it is hard to track if all cleanup tasks have been completed.

I schedule the work to do the cleanup also asynchronously which could mitigate this issue, lets me know if this solution is ok and if we aslo need to do this for init
PlatformMgr().ScheduleWork(HandleDisarmFailSafe);